### PR TITLE
Wip 13291 partition table

### PR DESCRIPTION
**Pull request by osynge**
Bug fix for http://tracker.ceph.com/issues/13291

### CODE REVIEW

File: src/ceph-disk [hunk 1]
```diff
@@ -527,6 +548,70 @@ def get_partition_dev(dev, pnum):
...
+
+    def blkid_udev_read(local_mode,blkid_udev_output):
+        for line in blkid_udev_output.splitlines():
+            (key, value) = line.split('=')
+            if key in output_set:
+                local_mode[key] = value
+
+    model = {}
+    mdl_blkid_udev_init(model)
+    blkid, _ = command(
```
Comment (dachary) [inline src/ceph-disk]: In some cases blkid does not work https://github.com/dachary/ceph/commit/9e796dc8b29c7f1a44d2aebcd0f59fa915270822. It can be repeated every time but I don't have a clue as to why it really happens. Using sgdisk works cross platform and reliably.

File: src/ceph-disk [hunk 2]
```diff
@@ -1719,6 +1804,9 @@ def main_prepare(args):
 
         if args.zap_disk is not None:
             zap(args.data)
```
Comment (dachary) [inline src/ceph-disk]: Great ! Could you make a separate pull request with this so it can be used to resolve http://tracker.ceph.com/issues/13291 ?

File: src/test/python/ceph_disk/test_ceph_disk_blikid.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    def test_all_devices_with_parition(self):
+        for device in self.all_devices:
+            if len(self.all_devices[device]) > 0:
+                status = ceph_disk.get_blkid_udev("/dev/" + device)
+                assert ceph_disk.view_blkid_udev_is_partioned(status) == True
+
+
+
+class TestGet_get_blkid_udev_with_loop_back(object):
```
Comment (dachary) [inline src/test/python/ceph_disk/test_ceph_disk_blikid.py]: unit tests run unprivileged and won't be able to use the loop device

File: src/ceph-disk (hunk 2 above)
Comment (dachary) [inline src/ceph-disk]: Do we need to zap a device that has no partition table ? In which case do we have a device with no partition table that could cause a problem if not zapped ?

File: src/test/python/ceph_disk/test_ceph_disk_blikid.py (hunk 3 above)
Comment (osynge) [inline src/test/python/ceph_disk/test_ceph_disk_blikid.py]: Fair enough, will need some help to make these integration tests though.

File: src/ceph-disk (hunk 2 above)
Comment (osynge) [inline src/ceph-disk]: The reason I wrote the patch in the first place, was that zeroed disks journals caused issues with 0.80 and 0.94 for journals, and also caused issues on 0.80 for data. Since this is no longer an issue for 0.94 I did not forward port this solution.

Comment (osynge) [inline src/ceph-disk]: The solution to the issue is to partition the disk.

Comment (osynge) [inline src/ceph-disk]: I have not checked that this issue exists on any platform other than SUSE SLE 12.

Comment (osynge) [inline src/ceph-disk]: Often "new" disks appear zeroed when you buy them.

**Comment by dachary** [comment]
Let's close this since there does not seem to be a problem after all.
